tests/test_pynq.py

```diff
@@ -0,0 +1,116 @@
+# Copyright Allo authors. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import tempfile
+import shutil
+
+import allo
+from allo.primitives.unify import unify
+from allo.ir.types import int8
+import warnings
+import numpy as np
+
+
+def _verify_pynq_project(project_dir, top_func_name=None):
+    """Verify PYNQ HLS scaffolding exists and contains basic expected headers.
+
+    - project_dir: path where run.tcl, kernel.cpp, kernel.h should live
+    - top_func_name: optional string of the top function name to check in kernel.cpp
+    """
+    run_tcl = os.path.join(project_dir, "run.tcl")
```

Comment:
You should also test whether your bd_tcl file exists, which is the most important and unique one to this pynq backend